common-rendering/src/components/keyEvents.tsx

```diff
@@ -0,0 +1,187 @@
+// ----- Imports ----- //
+
+import { css } from "@emotion/react";
+import type { SerializedStyles } from "@emotion/react";
+import { FC } from "react";
+import { textSans, headline } from "@guardian/src-foundations/typography";
+import { palette, remSpace } from "@guardian/src-foundations";
+import { SvgChevronUpSingle, SvgChevronDownSingle } from "@guardian/src-icons";
+import { Pillar, Theme } from "@guardian/types";
+import { from } from "@guardian/src-foundations/mq";
+
+// ----- Component ----- //
+type paletteId = 300 | 400;
+
+interface KeyEvent {
+	time: string;
```

Comment:
Is there more information on what format this data is sent as - is it an ISO date?

Reply:
Key event timestamps are always relative, i.e. will always be in the format `2m ago`/`2h ago`/`2d ago`. I guess you could argue that's a presentational concern, so we could pass in an absolute `Date` and run them through a function that produces a relative timestamp within the component? Actually that's probably sensible because we can include the absolute date in a `data` attribute on the element, and then we can update the relative timestamp over time on the client?

All that said I believe @marjisound was planning to handle the time stuff, and looking into the `timeAgo` lib, in a subsequent piece of work? So perhaps we ship this as a working piece of UI and update to provide the more complex time stuff a future PR?

Reply:
Yes if that's OK the plan is to handle the time in an upcoming PR by possibly using the time-ago library. 
It makes sense to make the time element format to be an absolute datetime and produce the relative timestamp within the component

Reply:
You mean the `time` string here is not an absolute time but is a string like '2m ago'?

If so, that sounds like it will need to be changed to be absolute otherwise it won't be possible to keep the content on the page live.

The `timeAgo` lib should probably be implemented using the [useTimeAgo](https://github.com/guardian/dotcom-rendering/pull/2708). Except.. This file has been deleted! I've no idea why that was done 😢 it used to be live on the page.

The `useTimeAgo` custom hook (when it isn't deleted) depends on the contract where `datetime` and `data-relativeformat` are present on a `time` element. `datetime` is already the semantic html contract for these elements, and `data-relativeformat` is a contract created in Frontend many years ago and what was used when `timeAgo` was built, so there's a lot of work gone into respecting this past work when adding this hook

Reply:
Thanks for the info! We have a separate ticket currently to investigate the timeAgo library & any accompanying implementations to use for liveblogs, but for the sake of building components now, we're going to go back and implement it slightly further on in the dev process!

Reply:
Thanks for the detailed explanation Oliver. I will add these details to the timeAgo ticket as guidance 👍